src/core/namespace.c

```diff
@@ -731,6 +731,107 @@ static int append_tmpfs_mounts(MountList *ml, const TemporaryFileSystem *tmpfs,
         return 0;
 }
 
+static int append_private_tmp(MountList *ml, const NamespaceParameters *p) {
+        MountEntry *me;
+
+        assert(ml);
+        assert(p);
+
+        /* When DynamicUser=yes enforce that /tmp/ and /var/tmp/ are disconnected from the host's
+         * directories, as they are world writable and ephemeral uid/gid will be used. */
```

Comment:
not your fault (mine actually), but this should be moved to unit_patch_contexts after PrivateTmp=disconnected is exposed

Reply:
There is already similar comments. I will remove this one.